<a href="https://colab.research.google.com/github/Jischak/Data-Analysis-of-Public-Ecommerce-in-Brazil/blob/main/notebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Data Analysis Project: Product Category Analysis of Public E-commerce in Brazil
- Name: Jischak Marthinus Tuny
- Email: tunyjischak@gmail.com
- ID linkedin: jiscmarth13

## Defining the Business Question

Case Study: Popular Product Categories on an E-commerce in Brazil

1. Which product categories do customers order the most?
2. Which product categories are the least ordered by customers?
3. What is the number of product categories ordered by region?
4. How many product categories are ordered by payment type?

## Prepare all required libraries

In [ ]:
import pandas as pd

#cek python version
#import sys; print(sys.version)

#check pandas library version
#pd.__version__

## Import All Dataset from Local

Make sure you have downloaded each of the required datasets from the following link:
* [Datasets](https://drive.google.com/drive/folders/1BeKeQYITva5C6Ynb3-0WoikPw0HXNVo1?usp=sharing)

Datasets that you need to upload:
1. customers_dataset.csv
2. order_items_dataset.csv
3. order_payments_dataset.csv
4. orders_dataset.csv
5. product_category_name_translation.csv
6. products_dataset.csv







In [ ]:
from google.colab import files

data_to_load = files.upload()

Saving customers_dataset.csv to customers_dataset.csv
Saving order_items_dataset.csv to order_items_dataset.csv
Saving order_payments_dataset.csv to order_payments_dataset.csv
Saving orders_dataset.csv to orders_dataset.csv
Saving product_category_name_translation.csv to product_category_name_translation.csv
Saving products_dataset.csv to products_dataset.csv


## Data Wrangling

### Gathering Data

In [ ]:
# Read the CSV file into a Pandas DataFrame
custs_df = pd.read_csv('customers_dataset.csv')
custs_df.head()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

In [ ]:
# Read the CSV file into a Pandas DataFrame
order_items_df = pd.read_csv('order_items_dataset.csv')
order_items_df.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

In [ ]:
# Read the CSV file into a Pandas DataFrame
order_payments_df = pd.read_csv('order_payments_dataset.csv')
order_payments_df.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [ ]:
# Read the CSV file into a Pandas DataFrame
orders_df = pd.read_csv('orders_dataset.csv')
orders_df.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

In [ ]:
# Read the CSV file into a Pandas DataFrame
product_cat = pd.read_csv('product_category_name_translation.csv')
product_cat.head()

product_category_name product_category_name_english
0            beleza_saude                 health_beauty
1  informatica_acessorios         computers_accessories
2              automotivo                          auto
3         cama_mesa_banho                bed_bath_table
4        moveis_decoracao               furniture_decor

In [ ]:
# Read the CSV file into a Pandas DataFrame
products_df = pd.read_csv('products_dataset.csv')
products_df.head()

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

### Assessing Data

In [ ]:
# check customers_dataset info
custs_df.info()

# check missing value of dataset
print("\nCheck Missing value")
print(custs_df.isna().sum())

# check duplicated value of dataset
print("\nCheck Duplicated value")
print(custs_df.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB

Check Missing value
customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

Check Duplicated value
0


customers_dataset evaluation:

1. Convert data type of columns
* customer_id               = object (ok)
* customer_unique_id        = object (ok)
* customer_zip_code_prefix  = int64 (ok)
* customer_city             = object (ok)
* customer_state            = object (ok)

In [ ]:
# check order_items_dataset info
order_items_df.info()

# check missing value of dataset
print("\nCheck Missing value")
print(order_items_df.isna().sum())

# check duplicated value of dataset
print("\nCheck Duplicated value")
print(order_items_df.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB

Check Missing value
order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

Check Duplicated value
0


order_items_dataset evaluation:

1. Convert data type of columns
* order_id             = object (ok)
* order_item_id        = int64 (ok)
* product_id           = object (ok)
* seller_id            = object (ok)
* shipping_limit_date  = object -> datetime
* price                = float64 (ok)
* freight_value        = float64 (ok)

In [ ]:
# check order_payments_dataset info
order_payments_df.info()

# check missing value of dataset
print("\nCheck Missing value")
print(order_payments_df.isna().sum())

# check duplicated value of dataset
print("\nCheck Duplicated value")
print(order_payments_df.duplicated().sum())

# check unique value of payment_type data
print("\nUnique value of payment_type")
print(order_payments_df["payment_type"].unique())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB

Check Missing value
order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

Check Duplicated value
0

Unique value of payment_type
['credit_card' 'boleto' 'voucher' 'debit_card' 'not_defined']


order_payments_dataset evaluation:

1. Convert data type of columns
* order_id              = object (ok)
* payment_sequential    = int64 (ok)
* payment_type          = object -> categorical
* payment_installments  = int64 (ok)  
* payment_value         = float64 (ok)

2. Boleto is Brazil's official payment method, regulated by the Central Bank of Brazil. In general, it is similar to a ticket/barcode. So, we will convert each boleto with a barcode.

In [ ]:
# check orders_dataset info
orders_df.info()

# check missing value of dataset
print("\nCheck Missing value")
print(orders_df.isna().sum())

# check duplicated value of dataset
print("\nCheck Duplicated value")
print(orders_df.duplicated().sum())

# check unique value of payment_type data
print("\nUnique value of orders status")
print(orders_df["order_status"].unique())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB

Check Missing value
order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_deliver

order_payments_dataset evaluation:

1. Convert data type of columns
* order_id                       = object (ok)
* customer_id                    = object (ok)
* order_status                   = object -> categorical
* order_purchase_timestamp       = object -> datetime
* order_approved_at              = object -> datetime
* order_delivered_carrier_date   = object -> datetime
* order_delivered_customer_date  = object -> datetime
* order_estimated_delivery_date  = object -> datetime

2. Change the NA values in some columns such as order_approved_at, order_delivered_carrier_date, and order_delivered_customer_date. Change the values to the known values afterwards.  

In [ ]:
# check product_category_name_translation dataset info
product_cat.info()

# check missing value of dataset
print("\nCheck Missing value")
print(product_cat.isna().sum())

# check duplicated value of dataset
print("\nCheck Duplicated value")
print(product_cat.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB

Check Missing value
product_category_name            0
product_category_name_english    0
dtype: int64

Check Duplicated value
0


In [ ]:
# check dataset info
products_df.info()

# check missing value of dataset
print("\nCheck Missing value")
print(products_df.isna().sum())

# check duplicated value of dataset
print("\nCheck Duplicated value")
print(products_df.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB

Check Missing value
product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g      

products_dataset evaluation:

1. Convert data type of columns
* product_id                   = object (ok)
* product_category_name        = object (ok)
* product_name_lenght          = float64 (ok)
* product_description_lenght   = float64 (ok)
* product_photos_qty           = float64 (ok)
* product_weight_g             = float64 (ok)
* product_length_cm            = float64 (ok)
* product_height_cm            = float64 (ok)
* product_width_cm             = float64 (ok)

2. Remove NA values in each columns.

### Cleaning Data

In [ ]:
# check dataset
order_items_df.info()

# convert shipping_limit_date data type from object to datetime
order_items_df["shipping_limit_date"] = pd.to_datetime( order_items_df["shipping_limit_date"] )

# After convertion
print("\nAfter convertion")
order_items_df.info()

# show dataset
print("\nShow dataset")
order_items_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB

After convertion
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype         
---  ------               --------------   -----         
 0   order_id             112650 non-null  object        
 1   order_item_id        112650 non-null  int64         
 2   prod

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

  shipping_limit_date   price  freight_value  
0 2017-09-19 09:45:35   58.90          13.29  
1 2017-05-03 11:05:13  239.90          19.93  
2 2018-01-18 14:48:30  199.00          17.87  
3 2018-08-15 10:10:18   12.99          12.79  
4 2017-02-13 13:57:51  199.90          18.14

In [ ]:
# check dataset info
order_payments_df.info()

# convert payment_type data type from object to categorical
payment_type = pd.CategoricalDtype(order_payments_df["payment_type"].unique())
order_payments_df["payment_type"] = order_payments_df["payment_type"].astype(payment_type)

#change value boleto to barcode
order_payments_df["payment_type"] = order_payments_df["payment_type"].replace(["boleto"], "barcode")

# After convertion
print("\nAfter convertion")
order_payments_df.info()

# show dataset
print("\nShow dataset")
order_payments_df.head()
print(order_payments_df.loc[ order_payments_df["payment_type"] == "barcode" ])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB

After convertion
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype   
---  ------                --------------   -----   
 0   order_id              103886 non-null  object  
 1   payment_sequential    103886 non-null  int64   
 2   payment_type          103886 non-null  category
 3   payment_installments  103886 non-null  int64   
 4   payment_value

In [ ]:
# check dataset info
orders_df.info()

# convert order_status data type from object to categorical
order_status = pd.CategoricalDtype(orders_df["order_status"].unique())
orders_df["order_status"] = orders_df["order_status"].astype(order_status)

# convert a few columns data type from object to datetime
orders_df["order_purchase_timestamp"] = pd.to_datetime( orders_df["order_purchase_timestamp"] )
orders_df["order_approved_at"] = pd.to_datetime( orders_df["order_approved_at"] )
orders_df["order_delivered_carrier_date"] = pd.to_datetime( orders_df["order_delivered_carrier_date"] )
orders_df["order_delivered_customer_date"] = pd.to_datetime( orders_df["order_delivered_customer_date"] )
orders_df["order_estimated_delivery_date"] = pd.to_datetime( orders_df["order_estimated_delivery_date"] )

# fill NA value
orders_df["order_approved_at"] = orders_df["order_approved_at"].fillna(method="ffill")
orders_df["order_delivered_carrier_date"] = orders_df["order_delivered_carrier_date"].fillna(method="ffill")
orders_df["order_delivered_customer_date"] = orders_df["order_delivered_customer_date"].fillna(method="ffill")


# After convertion
print("\nAfter convertion")
orders_df.info()

print("\nCheck Null Values")
print(orders_df.isna().sum())

# show dataset
print("\nShow dataset")
orders_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB

After convertion
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype         
---  ------                         --------------  -----         
 0

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp   order_approved_at  \
0    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37 2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49 2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06 2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39 2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0                    2017-10-18  
1                    2018-08-13  
2                    2018-09-04  
3                    2017-12-15  
4                    2018-02-26

In [ ]:
# check dataset info
products_df.info()

# Remove NAs value from dataset
print("\n", products_df.isna().sum())
print(products_df.shape)
products_df = products_df.dropna(axis=0, how="any")

# After remove NAs value
print("\nCheck Null Values")
print(products_df.isna().sum())
print(products_df.shape)

# show dataset
print("\nShow dataset")
products_df.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB

 product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

## Exploratory Data Analysis (EDA)

### Explore on customers_dataset

In [ ]:
# check summary descriptive statistic of dataset
print("\nSummary Descriptive Statistic")
print(custs_df.describe())

print("\nTotal Number of customers")
print(custs_df["customer_id"].nunique())

print("\nShow dataset")
custs_df.head()


Summary Descriptive Statistic
       customer_zip_code_prefix
count              99441.000000
mean               35137.474583
std                29797.938996
min                 1003.000000
25%                11347.000000
50%                24416.000000
75%                58900.000000
max                99990.000000

Total Number of customers
99441

Show dataset


customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

In [ ]:
# ANALYSIS: Top 10 Regions with the Highest Number of Customers
customers_based_on_regions = custs_df.groupby("customer_city")["customer_id"].nunique().reset_index().sort_values(by="customer_id",ascending=False)

print(customers_based_on_regions.head(10))

              customer_city  customer_id
3597              sao paulo        15540
3155         rio de janeiro         6882
453          belo horizonte         2773
558                brasilia         2131
1143               curitiba         1521
707                campinas         1444
2964           porto alegre         1379
3247               salvador         1245
1529              guarulhos         1189
3428  sao bernardo do campo          938


### Explore on orders_dataset

In [ ]:
# check summary descriptive statistic of dataset
print("\nSummary Descriptive Statistic")
print(orders_df.describe(datetime_is_numeric=True))

# Check status order
print("\nStatus order")
print(orders_df["order_status"].unique())

print("\nShow dataset")
orders_df.head()


Summary Descriptive Statistic
            order_purchase_timestamp              order_approved_at  \
count                          99441                          99441   
mean   2017-12-31 08:43:12.776581120  2017-12-31 20:23:34.460584704   
min              2016-09-04 21:15:19            2016-09-15 12:16:38   
25%              2017-09-12 14:46:19            2017-09-13 03:15:36   
50%              2018-01-18 23:04:36            2018-01-19 13:35:58   
75%              2018-05-04 15:42:16            2018-05-04 21:35:13   
max              2018-10-17 17:30:18            2018-09-03 17:40:06   

        order_delivered_carrier_date  order_delivered_customer_date  \
count                          99441                          99441   
mean   2018-01-05 01:00:33.728884480  2018-01-14 15:01:35.991602944   
min              2016-10-08 10:34:01            2016-10-11 13:46:32   
25%              2017-09-18 13:50:17            2017-09-26 00:48:43   
50%              2018-01-24 17:49:18         

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp   order_approved_at  \
0    delivered      2017-10-02 10:56:33 2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37 2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49 2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06 2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39 2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0                    2017-10-18  
1                    2018-08-13  
2                    2018-09-04  
3                    2017-12-15  
4                    2018-02-26

In [ ]:
# Merge orders_df dataset with custs_df
orders_df_merge = pd.merge(orders_df, custs_df, on='customer_id', how = 'inner')
orders_df_merge.head()

# Drop columns that not used
columns_used_orders_df_merge = ["order_id", "customer_id", "order_status", "customer_city"]
columns_not_used_orders_df_merge = [x for x in orders_df_merge.columns if x not in columns_used_orders_df_merge ]
orders_df_merge_minimal = orders_df_merge.drop(columns_not_used_orders_df_merge, axis=1)

# Check the unavailable orders
print(orders_df_merge_minimal.loc[ orders_df_merge_minimal["order_status"] == "unavailable"])

orders_df_merge_minimal.head()

                               order_id                       customer_id  \
266    8e24261a7e58791d10cb1bf9da94df5c  64a254d30eed42cd0e6c36dddb88adf0   
586    c272bcd21c287498b4883c7512019702  9582c5bbecc65eb568e2c1d839b5cba1   
687    37553832a3a89c9b2db59701c357ca67  7607cd563696c27ede287e515812d528   
737    d57e15fb07fd180f06ab3926b39edcd2  470b93b3f1cde85550fc74cd3a476c78   
1160   2f634e2cebf8c0283e7ef0989f77d217  7353b0fb8e8d9675e3a704c60ca44ebe   
...                                 ...                               ...   
99147  6cf5c427fb11bc5fe8abefa86aa7ba0b  91ac14a1e50bc022eeecf0c2ba37006a   
99233  1c7c8ab324927f5b6e52338bd110bef4  95b1767df51d6fa09402267d8de314df   
99252  aaab15da689073f8f9aa978a390a69d1  df20748206e4b865b2f14a5eabbfcf34   
99348  a69ba794cc7deb415c3e15a0a3877e69  726f0894b5becdf952ea537d5266e543   
99415  5fabc81b6322c8443648e1b21a6fef21  32c9df889d41b0ee8309a5efb6855dcb   

      order_status     customer_city  
266    unavailable  sao bento do sul

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status            customer_city  
0    delivered                sao paulo  
1    delivered                barreiras  
2    delivered               vianopolis  
3    delivered  sao goncalo do amarante  
4    delivered              santo andre

In [ ]:
# Filter success order = delivered
orders_df_merge_minimal_delivered = orders_df_merge_minimal.loc[ orders_df_merge_minimal["order_status"] == "delivered" ]

# ANALYSIS: Top 10 highest number of orders delivered by region
print("Top 10 orders by region")
top_10_order_by_region = orders_df_merge_minimal_delivered.groupby("customer_city")["order_id"].nunique().reset_index().sort_values(by="order_id", ascending=False)
print(top_10_order_by_region.head(10))

# ANALYSIS:  Top 10 lowest number of orders delivered by region
print("\nLowest 10 orders by region")
lowest_10_order_by_region = orders_df_merge_minimal_delivered.groupby("customer_city")["order_id"].nunique().reset_index().sort_values(by="order_id")
print(lowest_10_order_by_region.head(10))

Top 10 orders by region
              customer_city  order_id
3563              sao paulo     15045
3126         rio de janeiro      6601
449          belo horizonte      2697
553                brasilia      2071
1135               curitiba      1489
700                campinas      1406
2936           porto alegre      1342
3218               salvador      1188
1518              guarulhos      1144
3398  sao bernardo do campo       911

Lowest 10 orders by region
                customer_city  order_id
1005     conceicao do formoso         1
3188                 rubineia         1
3192                 sabaudia         1
1591                 ibitioca         1
1592                  ibitira         1
3195                 saboeiro         1
3196  sacra familia do tingua         1
1593                  ibitita         1
1594                 ibitiuva         1
3204     salinas da margarida         1


### Explore on product_category_name_translation

In [ ]:
# check summary descriptive statistic of dataset
print("\nSummary Descriptive Statistic")
print(product_cat.describe())

print("\nShow dataset")
product_cat.head()


Summary Descriptive Statistic
       product_category_name product_category_name_english
count                     71                            71
unique                    71                            71
top             beleza_saude                 health_beauty
freq                       1                             1

Show dataset


product_category_name product_category_name_english
0            beleza_saude                 health_beauty
1  informatica_acessorios         computers_accessories
2              automotivo                          auto
3         cama_mesa_banho                bed_bath_table
4        moveis_decoracao               furniture_decor

### Explore on products_dataset

In [ ]:
# check summary descriptive statistic of dataset
print("\nSummary Descriptive Statistic")
print(products_df.describe())

print("\nShow dataset")
products_df.head()


Summary Descriptive Statistic
       product_name_lenght  product_description_lenght  product_photos_qty  \
count         32340.000000                32340.000000        32340.000000   
mean             48.476592                  771.492393            2.188961   
std              10.245699                  635.124831            1.736787   
min               5.000000                    4.000000            1.000000   
25%              42.000000                  339.000000            1.000000   
50%              51.000000                  595.000000            1.000000   
75%              57.000000                  972.000000            3.000000   
max              76.000000                 3992.000000           20.000000   

       product_weight_g  product_length_cm  product_height_cm  \
count      32340.000000       32340.000000       32340.000000   
mean        2276.956586          30.854545          16.958813   
std         4279.291845          16.955965          13.636115   
min   

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

In [ ]:
# Change the product name in the dataset to English
products_df_merge = pd.merge(products_df, product_cat, on='product_category_name', how = 'left')
print(products_df_merge.head())

# Check the dataset info
print("\nCheck dataset info")
print(products_df_merge.info())

# Check the NA value of the dataset
print("\nCheck NA value of dataset")
print(products_df_merge.isna().sum())
products_df_merge.loc[ products_df_merge["product_category_name_english"].isna() == True ]

# Fill NA value in product_category_name_english column
products_df_merge["product_category_name_english"] = products_df_merge["product_category_name_english"].fillna("unknown_category")

# Check the dataset after filling each NA value
print("\nCheck dataset info")
print(products_df_merge.isna().sum())

# Remove the columns is not used in dataset
print("List columns in dataset")
print(products_df_merge.columns)
columns_used = ["product_id", "product_category_name_english"]
columns_not_used = [x for x in products_df_merge.columns if x not in columns_used ]
products_df_merge_minimal = products_df_merge.drop(columns_not_used, axis=1)


# Show the new dataset
products_df_merge_minimal.head()

                         product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  \
0             225.0               16.0               10.0              14.0   
1            1000.0     

product_id product_category_name_english
0  1e9e8ef04dbcff4541ed26657ea517e5                     perfumery
1  3aa071139cb16b67ca9e5dea641aaa2f                           art
2  96bd76ec8810374ed1b65e291975717f                sports_leisure
3  cef67bcfe19066a932b7673e239eb23d                          baby
4  9dc1a7de274444849c219cff195d0b71                    housewares

### Explore on order_items_dataset

In [ ]:
# check summary descriptive statistic of dataset
print("\nSummary Descriptive Statistic")
print(order_items_df.describe())

print("\nShow dataset")
order_items_df.head()


Summary Descriptive Statistic
       order_item_id          price  freight_value
count  112650.000000  112650.000000  112650.000000
mean        1.197834     120.653739      19.990320
std         0.705124     183.633928      15.806405
min         1.000000       0.850000       0.000000
25%         1.000000      39.900000      13.080000
50%         1.000000      74.990000      16.260000
75%         1.000000     134.900000      21.150000
max        21.000000    6735.000000     409.680000

Show dataset


order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

  shipping_limit_date   price  freight_value  
0 2017-09-19 09:45:35   58.90          13.29  
1 2017-05-03 11:05:13  239.90          19.93  
2 2018-01-18 14:48:30  199.00          17.87  
3 2018-08-15 10:10:18   12.99          12.79  
4 2017-02-13 13:57:51  199.90          18.14

In [ ]:
# Merge products_df_merge_minimal dataset with order_items_df
order_items_df_merge = pd.merge(order_items_df, products_df_merge_minimal, on='product_id', how = 'inner')
order_items_df_merge.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  130898c0987d1801452a8ed92a670612              1   
2  532ed5e14e24ae1f0d735b91524b98b9              1   
3  6f8c31653edb8c83e1a739408b5ff750              1   
4  7d19f4ef4d04461989632411b7e588b9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
2  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
3  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
4  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   

  shipping_limit_date  price  freight_value product_category_name_english  
0 2017-09-19 09:45:35   58.9          13.29                    cool_stuff  
1 2017-07-05 02:44:11   55.9          17.96                    cool_stuff  
2 2018-05-23 10:56:25   64.9          18.33                    cool_stuff  
3 2017-08-07 18:55:08   58.9          16.17                    cool_stuff  
4 2017-08-16 22:05:11   58.9          13.29                    cool_stuff

### Explore on order_payments_dataset

In [ ]:
# check summary descriptive statistic of dataset
print("\nSummary Descriptive Statistic")
print(order_payments_df.describe())

print("\nShow dataset")
order_payments_df.head()


Summary Descriptive Statistic
       payment_sequential  payment_installments  payment_value
count       103886.000000         103886.000000  103886.000000
mean             1.092679              2.853349     154.100380
std              0.706584              2.687051     217.494064
min              1.000000              0.000000       0.000000
25%              1.000000              1.000000      56.790000
50%              1.000000              1.000000     100.000000
75%              1.000000              4.000000     171.837500
max             29.000000             24.000000   13664.080000

Show dataset


order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [ ]:
# Merge order_items_df_merge dataset with order_payments_df
order_payments_df_merge = pd.merge(order_items_df_merge, order_payments_df, on='order_id', how = 'inner')
order_payments_df_merge.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  130898c0987d1801452a8ed92a670612              1   
2  532ed5e14e24ae1f0d735b91524b98b9              1   
3  6f8c31653edb8c83e1a739408b5ff750              1   
4  7d19f4ef4d04461989632411b7e588b9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
2  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
3  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
4  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   

  shipping_limit_date  price  freight_value product_category_name_english  \
0 2017-09-19 09:45:35   58.9          13.29                    cool_stuff   
1 2017-07-05 02:44:11   55.9          17.96                    cool_stuff   
2 2018-05-23 10:56:25   64.9          18.33                    cool_stuff   
3 2017-08-07 18:55:08   58.9          16.17                    cool_stuff   
4 2017-08-16 22:05:11   58.9          13.29                    cool_stuff   

   payment_sequential payment_type  payment_installments  payment_value  
0                   1  credit_card                     2          72.19  
1                   1      barcode                     1          73.86  
2                   1  credit_card                     2          83.23  
3                   1  credit_card                     3          75.07  
4                   1  credit_card                     4          72.19

In [ ]:
# create new dataset that will export
main_data = pd.merge(order_payments_df_merge, orders_df_merge_minimal, on='order_id', how = 'inner')
main_data.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  130898c0987d1801452a8ed92a670612              1   
2  532ed5e14e24ae1f0d735b91524b98b9              1   
3  6f8c31653edb8c83e1a739408b5ff750              1   
4  7d19f4ef4d04461989632411b7e588b9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
2  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
3  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
4  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   

  shipping_limit_date  price  freight_value product_category_name_english  \
0 2017-09-19 09:45:35   58.9          13.29                    cool_stuff   
1 2017-07-05 02:44:11   55.9          17.96                    cool_stuff   
2 2018-05-23 10:56:25   64.9          18.33                    cool_stuff   
3 2017-08-07 18:55:08   58.9          16.17                    cool_stuff   
4 2017-08-16 22:05:11   58.9          13.29                    cool_stuff   

   payment_sequential payment_type  payment_installments  payment_value  \
0                   1  credit_card                     2          72.19   
1                   1      barcode                     1          73.86   
2                   1  credit_card                     2          83.23   
3                   1  credit_card                     3          75.07   
4                   1  credit_card                     4          72.19   

                        customer_id order_status          customer_city  
0  3ce436f183e68e07877b285a838db11a    delivered  campos dos goytacazes  
1  e6eecc5a77de221464d1c4eaff0a9b64    delivered                  jatai  
2  4ef55bf80f711b372afebcb7c715344a    delivered         belo horizonte  
3  30407a72ad8b3f4df4d15369126b20c9    delivered   sao jose dos pinhais  
4  91a792fef70ecd8cc69d3c7feb3d12da    delivered   conselheiro lafaiete

In [ ]:
# ANALYSIS: Top 10 Category Products with The Highest Amount of Orders
top_10_highest_order_PC = main_data.groupby("product_category_name_english")["order_id"].nunique().reset_index().sort_values("order_id", ascending=False)
print("Top 10 Category Products with The Highest Amount of Orders")
print(top_10_highest_order_PC.head(10))

print("\nTotal Number of orders")
print(top_10_highest_order_PC["order_id"].sum())

# ANALYSIS: Top 10 Category Products with The Lowest Amount of Orders
print("\nTop 10 Category Products with The Lowest Amount of Orders")
top_10_lowest_order_PC = main_data.groupby("product_category_name_english")["order_id"].nunique().reset_index().sort_values("order_id")
print(top_10_lowest_order_PC.head(10))

Top 10 Category Products with The Highest Amount of Orders
   product_category_name_english  order_id
7                 bed_bath_table      9417
43                 health_beauty      8835
65                sports_leisure      7720
15         computers_accessories      6689
39               furniture_decor      6449
49                    housewares      5884
71                 watches_gifts      5624
68                     telephony      4199
5                           auto      3897
69                          toys      3886

Total Number of orders
98017

Top 10 Category Products with The Lowest Amount of Orders
   product_category_name_english  order_id
61         security_and_services         2
29     fashion_childrens_clothes         8
11             cds_dvds_musicals        12
52                    la_cuisine        13
70              unknown_category        22
3          arts_and_craftmanship        23
46                home_comfort_2        24
23           diapers_and_hygiene   

In [ ]:
print("\nTotal number of cities that placed an order")
print(main_data["customer_city"].nunique())

# ANALYSIS: Top 10 Number of Product Category Orders by Region
top_10_PC_by_region = main_data.groupby(["product_category_name_english", "customer_city"])["order_id"].nunique().reset_index("product_category_name_english").sort_values("order_id", ascending=False)
print("\nTop 10 Category Products with The Highest Amount of Orders")
top_10_PC_by_region.head(30)


Total number of cities that placed an order
4095

Top 10 Category Products with The Highest Amount of Orders


product_category_name_english  order_id
customer_city                                         
sao paulo                     bed_bath_table      1653
sao paulo                      health_beauty      1533
sao paulo                     sports_leisure      1266
sao paulo                         housewares      1107
sao paulo              computers_accessories      1004
sao paulo                    furniture_decor       998
sao paulo                      watches_gifts       778
rio de janeiro                bed_bath_table       726
sao paulo                               toys       610
sao paulo                          telephony       570
rio de janeiro                 health_beauty       523
rio de janeiro                sports_leisure       513
sao paulo                               auto       501
sao paulo                          perfumery       496
rio de janeiro               furniture_decor       483
rio de janeiro         computers_accessories       462
sao paulo                               baby       456
rio de janeiro                 watches_gifts       436
sao paulo                         cool_stuff       430
rio de janeiro                    housewares       412
sao paulo                       garden_tools       396
sao paulo                         stationery       348
sao paulo                        electronics       332
rio de janeiro                          toys       327
belo horizonte                bed_bath_table       297
sao paulo           fashion_bags_accessories       295
rio de janeiro                  garden_tools       267
sao paulo                           pet_shop       265
rio de janeiro                    cool_stuff       250
belo horizonte                 health_beauty       249

In [ ]:
total_orders_by_payment_type = main_data.groupby("payment_type")["order_id"].nunique().reset_index().sort_values("order_id", ascending=False)
print("Total Numbers of orders based on Payment Type")
print(total_orders_by_payment_type.head())

# ANALYSIS: Top 10 Number of Product Category Orders by Payment Type
top_10_PC_by_PT = main_data.groupby(["product_category_name_english", "payment_type"])["order_id"].nunique().reset_index("product_category_name_english").sort_values("order_id", ascending=False)
print("\nTop 10 Number of Product Category Orders by Payment Type")
top_10_PC_by_PT.head(30)

Total Numbers of orders based on Payment Type
  payment_type  order_id
0  credit_card     74914
1      barcode     19351
2      voucher      3708
3   debit_card      1494
4  not_defined         0

Top 10 Number of Product Category Orders by Payment Type


product_category_name_english  order_id
payment_type                                        
credit_card                 bed_bath_table      7540
credit_card                  health_beauty      6874
credit_card                 sports_leisure      5904
credit_card                furniture_decor      4919
credit_card          computers_accessories      4737
credit_card                     housewares      4613
credit_card                  watches_gifts      4568
credit_card                      telephony      3108
credit_card                           toys      3105
credit_card                           auto      2988
credit_card                     cool_stuff      2858
credit_card                   garden_tools      2603
credit_card                      perfumery      2501
credit_card                           baby      2287
credit_card                    electronics      1852
credit_card                     stationery      1829
barcode              computers_accessories      1723
barcode                      health_beauty      1704
barcode                     bed_bath_table      1601
barcode                     sports_leisure      1586
credit_card       fashion_bags_accessories      1392
credit_card                       pet_shop      1342
barcode                    furniture_decor      1339
barcode                         housewares      1044
barcode                          telephony       945
barcode                      watches_gifts       922
credit_card               office_furniture       897
credit_card            luggage_accessories       836
barcode                       garden_tools       814
credit_card                 consoles_games       806

## Export Dataset

In [ ]:
main_data.to_csv("main_data.csv", index=False)

## Visualization & Explanatory Analysis

Visualization and Explanatory Analysis can be seen on the Dashboard at the following link:

https://lookerstudio.google.com/reporting/96bb689b-fd99-419b-8c21-bc710fdda5f0

## Conclusion

- Conclusion Question 1: Products that are classified in the bed_bath_table category are the most ordered products with a total number of 9417.

- Conclusion Question 2: The least ordered product category is security & services with only 2 orders.

- Conclusion Question 3: In the top 10 product categories, the most orders came from the Sao Paoulo region. The other regions with the most orders were Rio de Janiero, Belo Horizonte, Brazillia, and Curitiba, respectively.

- Conclusion Question 4: The most popular payment type in the top 10 product categories is credit card, followed by several other payment methods such as barcodes, vouchers, and debit cards.